docs/src/components/ChunkedSnippet.astro

```diff
@@ -112,8 +120,24 @@ const { parser, lang, plugins } = getLanguageConfig(source) ?? {
   plugins: [],
 };
 
+// Strips the smallest common leading indentation from all lines
+const dedentBlock = (text: string): string => {
+  if (!text) return text
+  const lines = text.split('\n')
+  const nonEmpty = lines.filter((l) => l.trim().length > 0)
+  if (nonEmpty.length === 0) return text
+  const indents = nonEmpty.map((l) => (l.match(/^(\s*)/)?.[1].length ?? 0))
+  const minIndent = Math.min(...indents)
+  if (minIndent === 0) return text
+  return lines
+    .map((l) => (l.length >= minIndent ? l.slice(minIndent) : l.trimStart()))
+    .join('\n')
+}
+
 let output
-if (!parser || !lang || !rawOutput) {
+if (lang === 'rs') {
+  output = rawOutput ? dedentBlock(rawOutput) : 'Code snippet not found.'
```

Comment:
can we add this in the `else` block instead? to keep the language specific formatting there?